ruff.toml

```diff
@@ -33,11 +33,33 @@ line-length = 79
 indent-width = 4
 
 [lint]
-# Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
-# Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
-# McCabe complexity (`C901`) by default.
-select = ["E", "F", "W"]
-ignore = []
+preview = true  # this is needed for linting empty lines, etc.
+
+pydocstyle.convention = "google"
+
+select = ["ALL"]
+ignore = [
+    "PTH",
+    "ANN101",  # must be set in older version of ruff
+    "ANN204",
+    "COM812",
+    "S603",
+    "ERA001",
+    "TRY003",
+    "UP006",   # support for older python version (type vs. Type)
+    "UP007",   # ..^
+    "UP035",   # ..
+    "UP045",   # Use `X | None` for type annotations
+    "ARG002",
+    "INP001",  # add `__init__.py` to namespaced package
+    "FIX002",  # FIX all TODOs
+    "TD003",   # missing issue link
+    "S404",    # subprocess module is possibly insecure
+    "PLC0415", # import must be on top of the file
+    "CPY001",  # missing copyright header
+    "UP045"
+
```

Comment:
```suggestion
```

Reply:
This is connected to the ruff version. I think the current (added in this PR) needs this.